test/e2e_node/cpumanager_test.go

```diff
@@ -163,16 +166,36 @@ var _ = SIGDescribe("CPU Manager", ginkgo.Ordered, framework.WithSerial(), featu
 	})
 
 	ginkgo.BeforeEach(func(ctx context.Context) {
+		// note intentionally NOT set reservedCPUs -  this must be initialized on a test-by-test basis
 		podMap = make(map[string]*v1.Pod)
 	})
 
+	ginkgo.JustBeforeEach(func(ctx context.Context) {
+		// note intentionally NOT set reservedCPUs -  this must be initialized on a test-by-test basis
+
+		// use a closure to minimize the arguments, to make the usage more straightforward
+		skipIfAllocatableCPUsLessThan = func(node *v1.Node, val int) {
+			ginkgo.GinkgoHelper()
+			cpuReq := int64(val + reservedCPUs.Size()) // reserved CPUs are not usable, need to account them
+			// the framework is initialized using an injected BeforeEach node, so the
+			// earliest we can do is to initialize the other objects here
+			nodeCPUDetails := cpuDetailsFromNode(node)
+
+			msg := fmt.Sprintf("%v full CPUs (detected=%v requested=%v reserved=%v online=%v smt=%v)", cpuReq, nodeCPUDetails.Allocatable, val, reservedCPUs.Size(), onlineCPUs.Size(), smtLevel)
+			ginkgo.By("Checking if allocatable: " + msg)
+			if nodeCPUDetails.Allocatable < cpuReq {
+				e2eskipper.Skipf("Skipping CPU Manager test: not allocatable %s", msg)
```

Comment:
Skipping tests can potentially hide issues. Would it make sense failing the tests instead and ensure that CI machines running CPU tests have enough CPUs? In this case if CI configuration is changed again, the tests would start failing immediately, which is a clear signal.

Reply:
True. Problem is: it's a very delicate balance and I can't find good answers.
Making the test fail will create a storm of red tests which we can't fix but removing the relevant groups from the test lanes. Which brings the question: why did we add these tests in these lanes in the first place?
1. if we want the little signal we can get, then doing as many tests as we can and skipping others seems the closest to the intent
2. if we want the full suite running, then we can't. So either bump the machine specs or remove the relevant tests from the lanes.

Reply:
> Making the test fail will create a storm of red tests which we can't fix but removing the relevant groups from the test lanes.

I thought we can fix it by reverting test-infra changes that caused it.

Reply:
not sure we ever had test-infra changes which downsized the CI machines. 
But in general: can we just go and require bigger machines in CI? because this is ultimately the root cause (or, losing signal, remove the tests from CI runs).